### PR TITLE
Make Fill & Flip uncontrolled.

### PR DESCRIPTION
**Pull request by miina**
Related: #525 

This PR makes the Fill and Flip controls uncontrolled so that the state would be managed separately, only for the elements that actually use those values.

### CODE REVIEW

**Review by wassgha — DISMISSED** [review]
LGTM

File: assets/src/edit-story/components/panels/shared/flipControls.js [hunk 1]
```diff
@@ -18,38 +18,82 @@
...
+    ({ type }) => getDefinitionForType(type).canFlip
+  );
+
+  const updateProperties = useCallback(() => {
+    onSetProperties(({ flip: oldFlip, type }) => {
+      const update = {
+        ...state,
+        flip:
+          // Ensure flip change only if flip controls are actually visible (canFlip).
+          canFlip && getDefinitionForType(type).canFlip ? state.flip : oldFlip,
```
Comment (barklund) [inline assets/src/edit-story/components/panels/shared/flipControls.js]: `&& getDefinitionForType(type).canFlip` is this check necessary? Wouldn't that always be covered by `canFlip`?

**Review by barklund — COMMENTED** [review]
I like the idea of off-loading the updating of properties to individual components, but we're doing a lot of prop drilling here. I'm not sure this shallow of a rewrite is helping much.

Converting this setup to a context-based one would be a lot better, but again, that's also part of my concerns over in #526 - this whole `selectedElements`, `onSetProperties`, and `getCommonValue` needs a thorough refactor as it's pretty ugly right now.

I'd make it so that any individual component can do something like:

```js
const { actions: { getValue, updateValue }} = useSelection();
const isFill = getValue('isFill');
...
if (isFill === SELECTION_MULTIPLE) { ... }
...
<input
  onChange={(value) => updateValue('isFill', value)}
/>
...
```

It would be so much cleaner and easier to work with.

Some panels would need to know things like your `canFlip` here - we'd create utils in the provider for that. Other panels need to know how many elements are selected (e.g. for distribution or alignment), ditto for those.

Basically, we'd offload most of the repeated logic in all the panels to either the individual controls or the selection provider.

Then a panel would optimally just be:

```js
function SizePositionPanel() {
  return (
    <SimplePanel title="Size and position">
      <BackgroundButtonRow />
      <WidthHeightRow />
      <RotationRow />
    </SimplePanel>
  );
}
```

It would make life so much easier. It's a big ask and not just at thing for this PR, but I feel we're just patching tiny holes here instead of addressing the larger issue of a bad infrastructure for our current demands.

File: assets/src/edit-story/components/panels/shared/flipControls.js (hunk 1 above)
Comment (miina) [reply assets/src/edit-story/components/panels/shared/flipControls.js]: Indeed! Changed in a8cd9760

**Comment by miina** [comment]
@barklund Agreed that would be good to organize all the property settings instead of being all over the place. 
Def not going to do this in this PR, however, perhaps https://github.com/google/web-stories-wp/pull/412 could be a better candidate since that one is shuffling everything around a bit anyway.

**Comment by dvoytenko** [comment]
I think this change by itself could be ok. But I think we may benefit from a more high-viewer redo of side panels. @barklund is suggested a more context-based approach, and that could work well too. An additional note: the performance of the side panel seems to be progressively worsening. Which would also point to the need to reachitect it for some scale.

The one thing I keep noticing is the `useEffect(() => updateProperties(state.x), [state.x])` and that seems to be a causing all kind of problems for us. This could show that making controls more uncontrolled would dig this hole even deeper.

What if we had a contract where we'd have as an input `selectedElements` + `getCommonValues`. And as output, we'd be calling `pushPropertyUpdates(props, commit)` (or two methods `pushProperty` and `commit`). We will be pushing property updates in a fully controlled way, in-sync with inputs' `onChange` and `onBlur`. These updates would work locally across the side panel, but would be queued up before being pushed into the data model reducer. This will be akin to factoring `setState()` outside the panels and controls and make it, more or less, the exact replica of `updateElement()` protocol. It'd make the panels and controls themselves fully controlled, but with a clean layer for optimization.

**Comment by dvoytenko** [comment]
After saying all this, I realized that that'd make it more-or-less exactly the intended form protocol with fields and submit. So, possibly we could explore making our set up look even more like a form.

**Comment by miina** [comment]
Closing this in favor of #621 
cc @dvoytenko
